### PR TITLE
ICP-4808: Update curb energy monitor

### PR DESCRIPTION
**Pull request by munds**
This PR updates Curb Energy Monitor with these changes:

* All `atomicState` references have changed to `state`.
* Add the `usesThirdPartyAuthentication` tag for OneApp
* Add some debug statements
* Lets the user choose a location (and only one location)
* Fix an error in getting circuits
* Fix NPE by doing a check before sending events / executing commands
* Also fixes ICP-4808

The above might make this PR sound huge, but it's not :)

### CODE REVIEW

**Comment by munds** [comment]
Relates to: https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/2816

File: smartapps/curb/curb-energy-monitor.src/curb-energy-monitor.groovy [hunk 1]
```diff
@@ -128,18 +122,16 @@ def callback() {
         ]
         httpPostJson([uri: curbTokenUrl, body: tokenParams]) {
             resp ->
-                atomicState.refreshToken = resp.data.refresh_token
-              atomicState.authToken = resp.data.access_token
-
-            getCurbLocations()
+                state.refreshToken = resp.data.refresh_token
```
Comment (ron2911) [inline smartapps/curb/curb-energy-monitor.src/curb-energy-monitor.groovy:125]: indentation

File: smartapps/curb/curb-energy-monitor.src/curb-energy-monitor.groovy [hunk 2]
```diff
@@ -150,96 +142,105 @@ private removeChildDevices(delete) {
...
-            atomicState.locationNames = locationNameList
-            atomicState.locationLookup = locationLookup
+            state.locationNames = locationNameList
+            state.locationLookup = locationLookup
         }
     } catch (e) {
         log.error "something went wrong: $e"
     }
+    log.debug "Found locations: ${allLocations}"
+    allLocations
```
Comment (ron2911) [inline smartapps/curb/curb-energy-monitor.src/curb-energy-monitor.groovy:170]: are we missing a return statement here?

Comment (ron2911) [reply smartapps/curb/curb-energy-monitor.src/curb-energy-monitor.groovy:170]: Also, `getCurbLocations` is not being called anywhere? The only place where it is being called, it is commented out right now?

File: smartapps/curb/curb-energy-monitor.src/curb-energy-monitor.groovy [hunk 3]
```diff
@@ -286,7 +287,8 @@ def processDevices(resp, data) {
 }
 
 def processKwh(resp, data) {
-    if (resp.hasError()) {
+	log.debug "Processing billing data: ${resp.data}"
+    if (!isOK(resp)) {
```
Comment (ron2911) [inline smartapps/curb/curb-energy-monitor.src/curb-energy-monitor.groovy:291]: indent

File: smartapps/curb/curb-energy-monitor.src/curb-energy-monitor.groovy [hunk 4]
```diff
@@ -326,19 +330,21 @@ def toQueryString(Map m) {
...
             client_secret: appSettings.clientSecret,
-            refresh_token: atomicState.refreshToken
+            refresh_token: state.refreshToken
         ]
 
         httpPostJson([uri: curbTokenUrl, body: tokenParams]) {
             resp ->
-                atomicState.authToken = resp.data.access_token
+                state.authToken = resp.data.access_token
+                log.debug "Got authToken: ${state.authToken}"
```
Comment (ron2911) [inline smartapps/curb/curb-energy-monitor.src/curb-energy-monitor.groovy:347]: I am not sure what is the practice, but should we log tokens?

**Pull request closed without merging**